src/analysisd/decoders/plugins/json_decoder.c

```diff
@@ -225,10 +225,11 @@ static void readJSON (cJSON *logJSON, char *parent, Eventinfo *lf)
         if (logJSON->string) {
             if (parent) {
                 n = strlen(parent);
-                key = malloc(n + strlen(logJSON->string) + 2);
-                strcpy(key, parent);
-                key[n++] = '.';
-                strcpy(key + n, logJSON->string);
+                if (key = malloc(n + strlen(logJSON->string) + 2), key) {
```

Comment:
We should not continue if `malloc` returns `null`.
Please use:
```c
os_malloc(n + strlen(logJSON->string) + 2), key);
```

Reply:
Fixed